### PR TITLE
Add network create api test on CheckDuplicate

### PR DESCRIPTION
**Pull request by wenchma**
Signed-off-by: Wen Cheng Ma wenchma@cn.ibm.com

### CODE REVIEW

File: integration-cli/docker_api_network_test.go [hunk 1]
```diff
@@ -41,6 +41,31 @@ func (s *DockerSuite) TestApiNetworkCreateDelete(c *check.C) {
...
+		CheckDuplicate: true,
+	}
+	configNotCheck := types.NetworkCreate{
+		Name:           name,
+		CheckDuplicate: false,
+	}
+	// First check whether the network already exists or not
+	if isNetworkAvailable(c, name) {
+		// Creating another network with same name and CheckDuplicate must fail
+		createNetwork(c, configOnCheck, false)
```
Comment (runcom) [inline integration-cli/docker_api_network_test.go]: isn't this covered above? https://github.com/docker/docker/pull/18767/files#diff-d27869c471324416cf8914063c3ae8c3R36 (not sure)

File: integration-cli/docker_api_network_test.go [hunk 2]
```diff
@@ -41,6 +41,31 @@ func (s *DockerSuite) TestApiNetworkCreateDelete(c *check.C) {
...
+	configOnCheck := types.NetworkCreate{
+		Name:           name,
+		CheckDuplicate: true,
+	}
+	configNotCheck := types.NetworkCreate{
+		Name:           name,
+		CheckDuplicate: false,
+	}
+	// First check whether the network already exists or not
+	if isNetworkAvailable(c, name) {
```
Comment (runcom) [inline integration-cli/docker_api_network_test.go]: can't you just
```
id := createNetwork(c, configOnCheck, true)
c.Assert(isNetworkAvailable(c, name), checker.Equals, true)
```

Comment (wenchma) [inline integration-cli/docker_api_network_test.go]: @runcom good idea :+1:

**Comment by wenchma** [comment]
@runcom updated the PR, PTAL, thx. :smile:

File: integration-cli/docker_api_network_test.go [hunk 3]
```diff
@@ -41,6 +41,28 @@ func (s *DockerSuite) TestApiNetworkCreateDelete(c *check.C) {
...
+		Name:           name,
+		CheckDuplicate: false,
+	}
+
+	// Creating a new network first
+	createNetwork(c, configOnCheck, true)
+	isNetworkAvailable(isNetworkAvailable(c, name), checker.Equals, true)
+
+	// Creating another network with same name and CheckDuplicate must fail
+	createNetwork(c, configOnCheck, false)
```
Comment (runcom) [inline integration-cli/docker_api_network_test.go]: might be worth to remove the check here https://github.com/docker/docker/pull/18767/files#diff-d27869c471324416cf8914063c3ae8c3R36 cause you're now doing it here

File: integration-cli/docker_api_network_test.go [hunk 4]
```diff
@@ -41,6 +41,28 @@ func (s *DockerSuite) TestApiNetworkCreateDelete(c *check.C) {
...
+		CheckDuplicate: true,
+	}
+	configNotCheck := types.NetworkCreate{
+		Name:           name,
+		CheckDuplicate: false,
+	}
+
+	// Creating a new network first
+	createNetwork(c, configOnCheck, true)
+	isNetworkAvailable(isNetworkAvailable(c, name), checker.Equals, true)
```
Comment (runcom) [inline integration-cli/docker_api_network_test.go]: this should start with `c.Assert(...` :)

**Comment by wenchma** [comment]
@runcom thanks for your patient comments, :+1:   addressed.

**Comment by runcom** [comment]
LGTM if janky's happy :)

**Comment by duglin** [comment]
LGTM

**Pull request merged**